### PR TITLE
Generation of NGINX image and minor fixes

### PR DESCRIPTION
**Pull request by groeges**
Signed-off-by: Steven Groeger <groeges@uk.ibm.com>

### CODE REVIEW

File: scripts/nginx-ubi/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal
+
+LABEL name="kabanero-index" \
+      vendor="Kabanero" \
+      version="1" \
+      release="1" \
+      summary="Nginx container to host Kabanero collections" \
```
Comment (kilnerm) [inline scripts/nginx-ubi/Dockerfile]: Can we call these Enterprise stacks rather than Kabanero collections?

File: scripts/nginx-ubi/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal
+
+LABEL name="kabanero-index" \
+      vendor="Kabanero" \
```
Comment (kilnerm) [inline scripts/nginx-ubi/Dockerfile]: I think we should figure out a better vendor label.

File: scripts/nginx-ubi/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal
```
Comment (kilnerm) [inline scripts/nginx-ubi/Dockerfile:1]: We should remove all reference to Kabanero from this file.

File: scripts/nginx/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+version: '2'
```
Comment (kilnerm) [inline scripts/nginx/docker-compose.yml]: Again, we need to remove all reference to kabanero.

File: scripts/post_build.d/build_nginx.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
```
Comment (kilnerm) [inline scripts/post_build.d/build_nginx.sh:1]: Remove kabanero references.

File: scripts/tekton/README.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
+
```
Comment (kilnerm) [inline scripts/tekton/README.md]: Can we re-write this whole pipeline / doc to be generic.

File: scripts/tekton/collections-build-task.yaml [hunk 7]
```diff
@@ -0,0 +1,202 @@
+apiVersion: tekton.dev/v1alpha1
```
Comment (kilnerm) [inline scripts/tekton/collections-build-task.yaml]: Following the common theme can we remove collections and kabanero references.

File: scripts/tekton/openshift.yaml [hunk 8]
```diff
@@ -0,0 +1,55 @@
+---
```
Comment (kilnerm) [inline scripts/tekton/openshift.yaml]: Ditto re. Kabanero / collections

**Review by kilnerm — CHANGES_REQUESTED** [review]

File: scripts/nginx-ubi/Dockerfile (hunk 1 above)
Comment (groeges) [reply scripts/nginx-ubi/Dockerfile]: OK. Done.

File: scripts/nginx-ubi/Dockerfile (hunk 3 above)
Comment (groeges) [reply scripts/nginx-ubi/Dockerfile:1]: If we are not mentioning Kabanero then we will have to use something and we will need to pass this in as a variable. We will also need to use the same variable at other places in the hub_build.sh script.

File: scripts/post_build.d/build_nginx.sh (hunk 5 above)
Comment (groeges) [reply scripts/post_build.d/build_nginx.sh:1]: What do you suggest as an alternative - should this be "Appsody"

File: scripts/tekton/README.md (hunk 6 above)
Comment (groeges) [reply scripts/tekton/README.md]: Possibly, it was what Jarek wrote. Will take a look.

File: scripts/tekton/openshift.yaml (hunk 8 above)
Comment (groeges) [reply scripts/tekton/openshift.yaml]: Removed "collections"
What do you suggest as alternative for "kabanero"

File: scripts/nginx/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+FROM $NGINX_IMAGE
+
+COPY scripts/nginx/nginx.conf  /etc/nginx/nginx.conf
+COPY scripts/nginx/nginx-ssl.conf  /etc/nginx/nginx-ssl.conf
+COPY scripts/nginx/startup.sh  /opt/startup.sh
+COPY build/prefetch    /opt/www/public
+COPY assets            /opt/www/public
+COPY build/index-src   /opt/www/public
+
+#RUN rm /opt/www/public/*-local.yaml
```
Comment (kilnerm) [inline scripts/nginx/Dockerfile]: Can we delete this line?

**Review by kilnerm — DISMISSED** [review]
One minor gripe regarding a commented rather than deleted line.

We should look at adding some labels in the future but fine to omit for now.

File: scripts/nginx/Dockerfile (hunk 9 above)
Comment (groeges) [reply scripts/nginx/Dockerfile]: Yes, Done.

File: scripts/tekton/openshift.yaml (hunk 8 above)
Comment (groeges) [reply scripts/tekton/openshift.yaml]: Done

File: scripts/tekton/README.md (hunk 6 above)
Comment (groeges) [reply scripts/tekton/README.md]: Removed

File: scripts/tekton/collections-build-task.yaml (hunk 7 above)
Comment (groeges) [reply scripts/tekton/collections-build-task.yaml]: Done

File: scripts/post_build.d/build_nginx.sh (hunk 5 above)
Comment (groeges) [reply scripts/post_build.d/build_nginx.sh:1]: Done

File: scripts/nginx-ubi/Dockerfile (hunk 3 above)
Comment (groeges) [reply scripts/nginx-ubi/Dockerfile:1]: Done

**Review by kilnerm — APPROVED** [review]
LGTN

**Pull request merged**